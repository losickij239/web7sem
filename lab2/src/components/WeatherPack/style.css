.weather-list__li {
    list-style: none;
    margin-left: 0; /* Отступ слева в браузере IE и Opera */
    padding-left: 0;
}