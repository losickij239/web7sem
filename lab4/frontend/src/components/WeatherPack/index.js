import React from 'react'
import WeatherEntity from '../WeatherEntity'
import './style.css'


export default function WeatherPack({serverInfo}) {


    const data = [
        {
            type: 'Температура',
            value: serverInfo.main.temp + ' ˚C'
        },
        {
            type: 'Скорость ветра',
            value: serverInfo.wind.speed + ' м/с'
        },
        {
            type: 'Состояние неба',
            value: serverInfo.weather[0].description
        },
        {
            type: 'Давление',
            value: serverInfo.main.pressure + ' КПа'
        },
        {
            type: 'Влажность',
            value: serverInfo.main.humidity + ' %'
        }
    ];

    const weatherData = data.map(data =>
        <li key={data.type}><WeatherEntity data={data}/></li>
    );

    return (
        <ul className="weather-list__li">
            {weatherData}
        </ul>
    );

}
