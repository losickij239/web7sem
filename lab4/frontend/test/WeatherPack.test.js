import React from 'react';
import WeatherPack from '../src/components/WeatherPack';
import renderer from 'react-test-renderer';
import mock from '../src/fixtures'


it('отображается корректно', () => {
    const tree = renderer.create(
        <WeatherPack serverInfo={mock}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


