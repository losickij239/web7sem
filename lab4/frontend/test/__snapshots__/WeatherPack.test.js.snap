// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображается корректно 1`] = `
<ul
  className="weather-list__li"
>
  <li>
    <div
      className="alert alert-dark d-flex"
    >
      <div>
        Температура
      </div>
      <div
        className="ml-auto"
      >
        277.37 ˚C
      </div>
    </div>
  </li>
  <li>
    <div
      className="alert alert-dark d-flex"
    >
      <div>
        Скорость ветра
      </div>
      <div
        className="ml-auto"
      >
        2 м/с
      </div>
    </div>
  </li>
  <li>
    <div
      className="alert alert-dark d-flex"
    >
      <div>
        Состояние неба
      </div>
      <div
        className="ml-auto"
      >
        clear sky
      </div>
    </div>
  </li>
  <li>
    <div
      className="alert alert-dark d-flex"
    >
      <div>
        Давление
      </div>
      <div
        className="ml-auto"
      >
        1017 КПа
      </div>
    </div>
  </li>
  <li>
    <div
      className="alert alert-dark d-flex"
    >
      <div>
        Влажность
      </div>
      <div
        className="ml-auto"
      >
        86 %
      </div>
    </div>
  </li>
</ul>
`;
